core/types/block.go

```diff
@@ -474,6 +457,28 @@ func CalcUncleHash(uncles []*Header) common.Hash {
 	return rlpHash(uncles)
 }
 
+// CalcRequestsHash creates the block requestsHash value for a list of requests.
+func CalcRequestsHash(requests [][]byte) common.Hash {
```

Comment:
It can be rewrote to following to avoid intermediate hashes slice

```go
func CalcRequestsHash(requests [][]byte) common.Hash {
	h1, h2 := sha256.New(), sha256.New()
	buf := make([]byte, 0, 32)
	for _, item := range requests {
		h1.Reset()
		h1.Write(item)
		h2.Write(h1.Sum(buf[:]))
	}
	var result common.Hash
	h2.Sum(result[:0])
	return result
}
```

Reply:
This is a very clever suggestion!